### DISCUSSION TITLE
What happens if Remove method doesn't find the specified key?

### DISCUSSION THREAD

**Discussion by oasaleh** [ZiggyCreatures/FusionCache#309] [Q&A] [answered]
I looked up the docs and can't find anything about this behavior. The code, however, doesn't say it can throw an exception. Can someone link me to somewhere where I can find an answer? Thank you.

**Comment by jodydonetti** [comment, marked as answer]
Hi, `Remove` does not throw an exception if nothing is there for that key, as is common with any dictionary/hashmap structure, or cache in general.
Think of it as _"I want to make sure nothing is there"_.
Also, If you are using a setup with [L1+L2](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/CacheLevels.md) (memory level + distributed level), nothing may be there in memory on the node where you call it, but the action will be propagated to both the distributed level (L2) and to other nodes via the [backplane](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/Backplane.md).
Hope this helps.

**Reply by oasaleh** [reply]
Thank you. One more question, if I do this in program.cs file,
```
        builder.Services.AddFusionCache(Constants.LogsCacheName)
            .WithDefaultEntryOptions(opt => { opt.Duration = TimeSpan.FromDays(20); });
```
Does that mean each entry will have a unique counter that will expire it in 20 days? Or the entire cache will expire in 20 days from the day the 1st value is inserted?

**Reply by jodydonetti** [reply]
Each entry has its own duration.
The `DefaultEntryOptions` are just a way to set some defaults, but each entry is "isolated".
